docs/changelog/96161.yaml

```diff
@@ -4,3 +4,26 @@ area: "Search"
 type: enhancement
 issues:
  - 95541
+highlight:
+  title: Better indexing and search performance under concurrent indexing and search
+  body: "When a query like a match phrase query or a terms query targets a constant keyword field we can skip\
+  query execution on shards where the query is rewritten to match no documents. We take advantage of index mappings\
+  including constant keyword fields and rewrite queries in such a way that, if a constant keyword field does not\
+  match the value defined in the index mapping, we rewrite the query to match no document. This will result in the\
+  shard level request to return immediately, before the query is executed on the data node and, as a result, skipping\
+  the shard completely. Here we leverage the ability to skip shards whenever possible to avoid unnecessary shard\
+  refreshes and improve query latency. Avoiding such unnecessary shard refreshes improves query latency since the\
+  search thread does not need to wait anymore for unnecessary shard refreshes. Before introducing this change a query\
```

Comment:
Maybe also add after ` search thread does not need to wait anymore for unnecessary shard refreshes.`,  that searches that don't match the query will remain search idle and therefor the indexing throughput won't be impacted negatively by a refresh.